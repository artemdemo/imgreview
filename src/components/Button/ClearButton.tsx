import React from 'react';
import styled, { css } from 'styled-components';

export const clearButtonCss = css`
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;

  background: transparent;

  cursor: pointer;

  /* inherit font & color from ancestor */
  color: inherit;
  font: inherit;

  /* Normalize 'line-height'. Cannot be changed from 'normal' in Firefox 4+. */
  line-height: normal;

  /* Corrects font smoothing for webkit */
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;

  /* Corrects inability to style clickable 'input' types in iOS */
  -webkit-appearance: none;
`;

const ClearButton = styled.button`
  ${clearButtonCss}
`;

export default ClearButton;
