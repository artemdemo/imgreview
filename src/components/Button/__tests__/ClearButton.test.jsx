import React from 'react';
import renderer from 'react-test-renderer';
import ClearButton from '../ClearButton';

describe('ClearButton', () => {
    it('render empty', () => {
        const tree = renderer.create(
            <ClearButton />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
