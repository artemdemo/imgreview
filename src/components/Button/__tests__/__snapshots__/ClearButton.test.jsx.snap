// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton render empty 1`] = `
<button
  className="sc-bdVaJa hmFSRG"
/>
`;
